src/dataflow/src/render/reduce.rs

```diff
@@ -213,24 +248,42 @@ where
                             *expected_group_size,
                         );
                         to_collect.push(collection.as_collection(move |key, val| {
-                            (key.clone(), (Some(index), val.clone()))
+                            (
+                                key.clone(),
+                                (ReductionType::Other, Some(index), val.clone()),
+                            )
                         }));
                     }
                     let is_accumulable = aggregates
                         .iter()
                         .map(|a| accumulable_hierarchical(&a.func).0)
                         .collect::<Vec<_>>();
+                    let is_min_max = aggregates
+                        .iter()
+                        .map(|a| is_min_or_max(&a.func) && !monotonic)
+                        .collect::<Vec<_>>();
+
                     differential_dataflow::collection::concatenate(scope, to_collect)
                         .reduce_abelian::<_, OrdValSpine<_, _, _, _>>("ReduceCollation", {
                             let mut row_packer = RowPacker::new();
                             move |key, mut input, output| {
-                                // The inputs are pairs of an optional index and row to decode.
+                                // The inputs are triples of a reduction type, an optional index and row to decode.
 
                                 // There can be at most one `None` index, and it indicates the accumulable aggregates.
                                 let new_row = Row::new(Vec::new());
-                                let mut accumulable = if (input[0].0).0 == None {
+                                let mut accumulable = if (input[0].0).0 == ReductionType::FusedAccumulable {
                                     // This input corresponds to our densely packed accumulable aggregates.
-                                    let iter = (input[0].0).1.iter();
+                                    let iter = (input[0].0).2.iter();
+                                    // Make sure we never try to read from this input again.
+                                    input = &input[1..];
+                                    iter
+                                } else {
+                                    new_row.iter()
+                                };
+
+                                let mut min_max = if (input[0].0).0 == ReductionType::FusedMinMax {
```

Comment:
I think something weird is going on here if we have both a `ReductionType::FusedMinMax` and a `ReductionType::FusedAccumulable` in our input. Previously the `None` would be the first element and it could have only one meaning. Now we might have two things, but we are only looking at the first position.

Reply:
I think this works because no matter what, if we have both, we would see `FusedAccumulable` before `FusedMinMax` and in that case I think line 278 forces us to remove the first element after we observe it to be accumulable. I haven't tested this case specifically and can do so in a bit